.github/workflows/windows-builds.yml

```diff
@@ -0,0 +1,142 @@
+name: 'Build on Windows'
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - '**.md'
+  release:
+    types:
+      - published
+
+jobs:
+  win64:
+    name: 'Windows 64-bit'
+    runs-on: [windows-latest]
+    env:
+      QT_VERSION: '5.12.9'
+      CMAKE_GENERATOR: "Visual Studio 16 2019"
+    steps:
+      - name: 'Add msbuild to PATH'
+        uses: microsoft/setup-msbuild@v1.0.2
+      - name: 'Checkout'
+        uses: actions/checkout@v2.3.3
+        with:
+          submodules: 'recursive'
+      - name: 'Get Cockatrice git info'
+        shell: bash
+        working-directory: ${{ github.workspace }}
+        run: |
+          git fetch --prune --unshallow
+          echo "GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)" >> $GITHUB_ENV
+          echo "GIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
```

Comment:
isn't this already available as GITHUB_SHA?

Reply:
GITHUB_SHA is not the same when dealing with PRs vs the main repo, apparently: <https://github.community/t/github-sha-not-the-same-as-the-triggering-commit/18286>

This is safer from what I understand. There's a couple other issues I've seen on the checkout actions repo as well referencing this. Doing it this way will ensure that the triggering PR hash is being used.